src/AWS.Deploy.Common/Recipes/IRecipeHandler.cs

```diff
@@ -24,5 +25,10 @@ public interface IRecipeHandler
         /// other locations that are monitored by the same source control root as the target application that needs to be deployed.
         /// </summary>
         Task<HashSet<string>> LocateCustomRecipePaths(string targetApplicationFullPath, string solutionDirectoryPath);
+
+        /// <summary>
+        /// Runs the recipe level validators and returns a list of failed validations
+        /// </summary>
+        List<ValidationResult> RunRecipeValidators(Recommendation recommendation, IDeployToolValidationContext deployToolValidationContext);
```

Comment:
This addition feels a little odd here. It's not documented at the top, but this interface/class focuses on loading recipe definitions prior to selecting a deployment option relatively early in the workflow. Then this runs relatively relatively later once we've selected a recipe and option settings, and doesn't share any of the other manager/handlers that were already present in `RecipeHandler`.

That said - I don't really have a better suggestion. Even though `OptionSetting` is in the name, `IOptionSettingHandler` does have `RunOptionSettingValidators` which is pretty close to this.

Though do like that this is consolidating the recipe validator execution between CLI and server mode.

----

(ultimately not a blocker, but curious what any other reviewers think)